main.nf

```diff
@@ -45,88 +82,696 @@ workflow.onComplete {
+	ch_stranger_meta = ch_samplesheet
+		.map { row -> // TODO: Rename this channel
+			tuple(
+				row.group,
+				row.id,
+				row.sex,
+				row.mother,
+				row.father,
+				row.phenotype,
+				row.diagnosis,
+				row.type,
+				row.assay,
+				row.clarity_sample_id,
+				(row.containsKey("ffpe") ? row.ffpe : false),
+				(row.containsKey("analysis") ? row.analysis : false)
+			)
+		}
+		.filter { row ->
+			def type = row[7]
+			type == "proband"
+		}
+
+	// meta sent to create_yml
+	ch_scout_yaml_meta = ch_samplesheet.map { row ->
+		tuple(
+			row.group,
+			row.id,
+			row.diagnosis,
+			row.assay,
+			row.type,
+			row.clarity_sample_id,
+			(row.containsKey("analysis") ? row.analysis : false)
+		)
+	}
+	.filter { row ->
+			def type = row[4]
+			type == "proband"
+	}
+
+	ch_gatkcov_meta = ch_samplesheet.map { row ->
+		tuple(
+			row.group,
+			row.id,
+			row.sex,
+			row.type,
+		)
+	}
+
+	ch_expansionhunter_meta = ch_gatkcov_meta.filter { it ->
+		it[3] == "proband"
+	} // ch group, id, sex, type
+	ch_svvcf_to_bed_meta = ch_expansionhunter_meta
+
+	// BAM-start
+	// Check for .bam files in read1 and start from bam if any found.
+	ch_bam_start = ch_samplesheet
+		.filter {
+			row -> row.read1.endsWith("bam")
+		}
+		.map {
+			row ->
+			def group = row.group
+			def id = row.id
+			def bam = row.read1
+			def bai = row.read2
+			tuple(group, id, bam, bai)
+		}
+
+
+	copy_bam(ch_bam_start)
+	bamtoyaml(ch_bam_start)
+	ch_output_info = ch_output_info.mix(bamtoyaml.out.bamchoice_INFO)
+
+	ch_bam_start_dedup_dummy = Channel.empty()
+	if(params.run_melt) {
+		dedupdummy(ch_bam_start)
+		ch_bam_start_dedup_dummy = dedupdummy.out.dedup_dummy
+	}
+
+	ch_bam_bai = Channel.empty()
+	ch_bam_bai = ch_bam_bai.mix(copy_bam.out.bam_bai)
+
+	// PED //
+	ch_ped_input = ch_samplesheet
+		.filter { row -> row.type == "proband" }
+		.map { row ->
+			def group = row.group
+			def id = row.id
+			def type = row.type
+			def sex = row.sex
+			def father = row.father
+			def mother = row.mother
+			tuple(group, id, type, sex, mother, father)
+			}
+
+	create_ped(ch_ped_input)
+	ch_ped_base = create_ped.out.ped_base
+	ch_ped_fa = Channel.empty()
+	ch_ped_ma = Channel.empty()
+
+	ch_ped_trio = Channel.empty()  //TODO: better name for this channel
+	ch_ped_trio = ch_ped_trio.mix(ch_ped_base)
+	if(params.mode == "family" && params.assay == "wgs") {
+
+		ch_ped_fa = create_ped.out.ped_fa
+		ch_ped_ma = create_ped.out.ped_ma
+
+		ch_ped_trio = ch_ped_trio.mix(ch_ped_fa).mix(ch_ped_ma)
+		madeline(ch_ped_trio)
+
+		ch_versions = ch_versions.mix(madeline.out.versions.first())
+		ch_output_info = ch_output_info.mix(madeline.out.madde_INFO)
+
+	}
+
+	// FASTQ //
+	if (params.umi) {
+		fastp(ch_fastq)
+		ch_fastq = fastp.out.fastq_trimmed_reads
+		ch_versions = ch_versions.mix(fastp.out.versions.first())
+	}
+
+	// ALIGN //
+	//TODO: why do we have a params.align conditional anyway?
+	ch_dedup_stats = Channel.empty()
+	if (params.align) {
+		bwa_align(ch_fastq)
```

Comment:
nf-core pipelines tend to have processes and subworkflows written in CAPITAL_SNAKES

A bit screamy, but makes it easier to distinguish things in the code I think. What do you think?